docs/book/start/parsing-packets.md

```diff
@@ -0,0 +1,100 @@
+# Parsing Packets
+
+In the previous chapter, our XDP application ran until Ctrl-C was hit and permitted
+all the traffic. Each time a packet was received, the BPF program created a logged
+the string "received a packet" for each packet received. In this chapter we're
+going to show how to parse packets.
+
+While we could go all out and extract data all the way up to L7, we'll constrain
+our example to L3, and to make things easier, IPv4 only.
+
+!!! example "Source Code"
+
+    Full code for the example in this chapter is available [here](https://github.com/aya-rs/book/tree/main/examples/xdp-log)
+
+## Using Network Types
+
+To get useful data to log, we first need some useful data structures to
```

Comment:
We're going to log the source ip address of incoming packets. So we'll need to: 

- Read the ethernet header to determine if we're dealing with an IPv4 packet, else terminate parsing
- Read the source IP Address from the IPv4 header

We could read the specifications of those protocols and parse manually, but instead we're going to use the network-types crate which provides convenient type definitions for many of the common internet protocols.

Let's add it to ...